### DISCUSSION TITLE
What if CopperSpice copy code from you?

### DISCUSSION THREAD

**Discussion by ghost** [RolandHughes/copperspice#15] [General]
The license is compatible. And you can't change the license.

**Comment by RolandHughes** [comment]
I don't care. Given all of the branding changes, other than a one line fix to something, it would be an awful lot of work for them, but OpenSource means just that, **OpenSource**.

There is actually some legal threshold, which I do not remember off the top of my head, about how much code one can steal without even posting a comment in the code stating where it came from.

As this progresses, there will be incredibly small overlap. Most of the stuff is getting nuked/replaced. Using GLFW only for the graphics layer will allow not only for the nuking of all "platform graphics" support but also allow the widget core code to directly use GLFW API instead of wrapper calls.

**Comment by RolandHughes** [comment]
One last follow-up on this. New modules having only a Roland Hughes/Logikal Solutions or Basis Doctrina copyright will be released under the Basis Doctrina License.

> Basis Doctrina License
> 
> Basis Doctrina is free software. You can redistribute it and/or modify it under
> the terms of this license.
> 
> Basis Doctrina is distributed in the hope that it will be useful, but WITHOUT
> ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
> FOR A PARTICULAR PUPOSE.
> 
> The Basis Doctrina License is a restricted GNU Lesser General Public License
> version 2.1 as published by the Free Software Foundation.
> https://www.gnu.org/licenses/
> 
> The restriction is that you may not include any part of this software in a
> software library that you charge money for either by commercial license, or
> mandatory/required donation, or subscription of any kind. The perversion of
> OpenSource to mean "everyone must buy a license" ends here.
> 
> This in no way restricts commercial products (software or hardware) that
> actually "do" something. It only applies to libraries and other SDKs. YOU CANNOT
> CHARGE MONEY FOR THEM IN ANY WAY.
>
